www/js/splash/notifScheduler.js

```diff
@@ -200,7 +250,7 @@ angular.module('emission.splash.notifscheduler',
             return;
         }
         setUpActions();
-        update();
+        update.then();
```

Comment:
Can you add more explanation for what this does?
I read the commit message but still don't see why we don't just call `update()` here

Reply:
Just calling `update()` would mean that the "ready" method would return without waiting for the promise that `update()` returns to resolve. If we don't wait here, we have the same messiness as before with the notifications not being scheduled before we get them to populate the table

Reply:
If you change it, do you actually observe a difference? Even if you do, I don't think it's for the reason you think it is.

Adding a `.then()` block doesn't block the current scope from continuing. It attaches a function to be called asynchronously once the promise resolves.
The block inside `ready().then(...)` is going to finish executing whether you have a `.then` or not.

Secondly, none of the code here is determining when `ready` returns.
It's just for declaring the things you want to do *after* Ionic is ready and the app config is loaded.

Reply:
That makes sense, I think making that change must have been a lucky coincidence that the loading timed out right the few times I tried it then. Taking it out leaves a situation where sometimes it loads in the right timing and sometimes does not. 

I think I'm going to have to find some condition to cause a refresh once that data is ready, kind of like the `appConfig` worked well for the sync and collect settings. I have `refreshNotifications` called on 'appConfig' now, but I think it's too soon since it sounds like that's the same time the `ready` in `notifScheduler` starts

Reply:
I ended up refactoring the way that `getScheduledNotifs` works so that if `isScheduling` (an existing boolean in `NotificationScheduler` was true, it used `scheduledPromise.then(*code to get notifications*)` to ensure the scheduling was completed before the fetching. This works consistently. 

It does feel a little clunky to me, and I'm a little worried that it might slow things down (though I haven't noticed that yet) so I'm definitely open to feedback on better ways to ensure the notifs are displayed consistently. I also think this is something that might get smoothed out once we're fully in React.